### DISCUSSION TITLE
Delete Firebase users by email id

### DISCUSSION THREAD

**Discussion by Dk201833** [invertase/react-native-firebase#6303] [General]
Can we delete the firebase user from app if the  user is not signed in or if user id signin to app.

**Comment by mikehardy** [comment]
Hi there! This scenario is not very detailed, so I'm not sure exactly what you want. However if `auth().currentUser !== undefined` then you have a current user session, and you can always call `auth().currentUser.signOut()`. That will invalidate the local user session on device, which is the same as "deleting the user" locally.

If you mean to delete the user completely from the firebase backend as well, you either need to have a currentUser session that is valid (and as a best practice it is typical to re-authenticate them prior to delete, in fact I believe the auth module makes you do so if the session is not very fresh) then you call this https://rnfirebase.io/reference/auth/user#delete - if they are not signed in then you need to make some other sort of flow using cloud functions so you can access the firebase-admin-sdk where you can manipulate auth users via an admin token, regardless of auth user token availability

**Comment by Rolika-agarwal** [comment]
I want to delete a user from firebase by email. Is it possible?

**Reply by mikehardy** [reply]
My answer above is still authoritative for your question
